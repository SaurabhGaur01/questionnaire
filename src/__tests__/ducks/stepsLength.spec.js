import {
    setStepsLength,
    stepsLengthReducer
} from '../../ducks/stepsLength';

describe('stepsLength Duck Tests', () => {
    const mockState = 1;

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setStepsLength should match the snapshot' , () => {
            expect(setStepsLength()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(stepsLengthReducer()).toEqual(0);
        });
        it('should return the passed state if called with no action' , () => {
            expect(stepsLengthReducer(mockState)).toBe(mockState);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(stepsLengthReducer(mockState, unknownAction)).toBe(mockState);
        });
    });

    describe('when called with a particular action', () => {
        it('should return the value which is passed' , () => {
            expect(stepsLengthReducer('', setStepsLength()))
            .toEqual(5);
        });
    });

})