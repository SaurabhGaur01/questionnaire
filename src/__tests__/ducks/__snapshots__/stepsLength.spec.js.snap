// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stepsLength Duck Tests Action Generator setStepsLength should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/SET_STEPS_LENGTH",
}
`;
